sensor_msgs/src/sensor_msgs/point_cloud2.py

```diff
@@ -150,28 +150,35 @@ def create_cloud(header, fields, points):
     @type  points: list of iterables, i.e. one iterable for each point, with the
                    elements of each iterable being the values of the fields for 
                    that point (in the same order as the fields parameter)
+    @param height: the height of the point cloud [default: 1]
+    @type height: unsigned int
+    @param width: the width of the point cloud [default: None]
+    @param width: unsigned int
     @return: The point cloud.
     @rtype:  L{sensor_msgs.msg.PointCloud2}
     """
+    if height == 1:
+        width = len(points)
 
     cloud_struct = struct.Struct(_get_struct_fmt(False, fields))
 
     buff = ctypes.create_string_buffer(cloud_struct.size * len(points))
 
-    point_step, pack_into = cloud_struct.size, cloud_struct.pack_into
+    point_step = cloud_struct.size
```

Comment:
These intermediate values appear to only be used once so are probably not worth creating at all.